homeassistant/components/camera/onvif.py

```diff
@@ -40,13 +56,59 @@
     vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
 })
 
+SERVICE_PTZ_SCHEMA = vol.Schema({
+    ATTR_ENTITY_ID: cv.entity_ids,
+    ATTR_PAN: vol.In([DIR_LEFT, DIR_RIGHT]),
+    ATTR_TILT: vol.In([DIR_UP, DIR_DOWN]),
+    ATTR_ZOOM: vol.In([ZOOM_OUT, ZOOM_IN])
+})
+
 
 @asyncio.coroutine
 def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     """Set up a ONVIF camera."""
     if not hass.data[DATA_FFMPEG].async_run_test(config.get(CONF_HOST)):
         return
-    async_add_devices([ONVIFCamera(hass, config)])
+
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
+    entities = []
+
+    def handle_ptz(service):
+        """Handle PTZ service call."""
+        tilt = service.data.get(ATTR_TILT, None)
```

Comment:
Is this check and `upper` call needed? I've wrote it before creating a schema for service. With schema it shouldn't be necessary. Am I right?